maintainers-guide: refresh (#6397)
tldr-pages/tldr: contributing-guides/maintainers-guide.md

> - PRs should be merged once they

Reviewer: ```suggestion
- PRs should be merged once they pass all following points:
```

Author: I disagree.

Reviewer: do you disagree with the content or the wording? It makes a previously implicit point explicit, reducing reader friction

Reviewer 2: I disagree with the content. I think it interrupts the reader's flow a bit.

  (1) **pass the automated tests** (GitHub Actions, CLA signing, etc.),

>   (2) have the **review comments addressed**,

Reviewer: ```suggestion
  (2) all **review comments are addressed**
```

Reviewer 2: This would only make sense with the change above, so I'll resolve this to keep the conversation cleaner.

>   (3) get **approved reviews by two maintainers**, (the second maintainer can merge immediately after approving) and

Reviewer: ```suggestion
  (3) are **approved by two maintainers**
      (the second maintainer can merge immediately after approving)
```

Reviewer 2: I agree with `get approved reviews` → `are approved`, but I don't understand the newline.

Reviewer: better readability. On my screen at least I cannot see the whole line in your proposal.
https://sembr.org

  (4) have been open for at least **24 hours** unless the changes are trivial

- If a PR fails to get a review from a second maintainer after a few days,

>   the first maintainer should ping others for review.

Reviewer 2: As discussed above, I think we should revert this change.
```suggestion
  the first maintainer should ping others for review. If it still lingers around
  for **over a week without a second maintainer’s approval**,
  the first maintainer can go ahead and merge it.
```

  (1) **pass the automated tests** (GitHub Actions, CLA signing, etc.),
  (2) have the **review comments addressed**, and
  (3) get **approved reviews by two maintainers**

>   (4) are at least **24 hours** old unless changes are trivial

Author: Feel free to change the value, I've been using 24 hours but that is just me.

Reviewer 3: I would suggest that both of these items are good to have here.

>   (2) have the **review comments addressed**, and
>   (3) get **approved reviews by two maintainers**
>   (4) are at least **24 hours** old unless changes are trivial

Reviewer 2: I think 24 hours is a good rule of thumb. I'd rephrase this a little though.
```suggestion
  (2) have the **review comments addressed**,
  (3) get **approved reviews by two maintainers**, and
  (4) have been open for at least **24 hours** unless the changes are trivial
```

Reviewer 4: Looks like Sbrl keep the explanation that
> the second maintainer can perform the merge immediately after accepting.
Leaving explicit that after the 2nd approve the maintainer shouldn't wait another stuff.

Reviewer 2: @sbrl reffering to your comment [here](https://github.com/tldr-pages/tldr/pull/6397#discussion_r693549872), what do you think about this:
```suggestion
  (2) have the **review comments addressed**,
  (3) get **approved reviews by two maintainers**, (the second maintainer can merge immediately after approving) and
  (4) have been open for at least **24 hours** unless the changes are trivial
```

Reviewer 2: @CleanMachine1 could you please apply this? ⬆️

Author: @marchersimon Could you please resend the suggestion or feel free to edit the page, I can't see whats being changed.

Reviewer 2: Done in ba415df

- If a PR fails to get a review from a second maintainer after a few days,

>   the first maintainer should ping others for review.

Reviewer 3: This should stay as-is.

Author: PRs need 2 approvals so it is wrong.

Reviewer 4: @CleanMachine1 This clause gives the power to org owners to force merge skipping that PR check.

Reviewer 2: But should we (you) even do that? We've never had a PR which had to be merged that urgently but didn't have 2 approvals.

Reviewer 2: So what should we do about this? Leave it?

Author: Maybe it can be phrased to show that only in occasions such as bug testing and trivial changes would be acceptable

Reviewer 2: I think even trivial changes should be approved by at least two maintainers. The rule isn't really relevant, so I think we can just leave it.

Author: @sbrl could we get your reply about this subject.

Reviewer 3: > But should we (you) even do that? We've never had a PR which had to be merged that urgently but didn't have 2 approvals.

Urgently? No, not really - usually posting urgent urgent urgent ones in Gitter is enough. There have been plenty of times when I've force-merged for example an obvious typo though. Generally speaking I only force-merge uncontroversial changes in periods of very high activity to avoid too much notification spam.

Reviewer 2: Hm, I never noticed that, but it makes sense. I think we can keep the original, but I don't have a strong opinion here.

Reviewer: I would side with the modification, this practice does not concern typical maintainers and we should strive to cut down noise.

Reviewer 3: I'm not sure what you mean by "typical maintainer" here @xeruf. Note that the maintainer's guide should be the definitive guide for *all* maintainers - regardless of experience level.

- If the only issues holding up a merge are **trivial fixes**
  (typos, syntax errors, etc.), and the author doesn't respond in a day or two,
  **maintainers can make the necessary changes themselves**,
  and proceed with the merge process.

- If a PR **stops getting feedback from the submitter** and is marked as stale

>   by [probot-stale](../.github/workflows/stale.yml),

Author: This error was just because it was outdated.
